packages/web-app-files/tests/unit/components/SideBar/Actions/FileActions.spec.js

```diff
@@ -0,0 +1,245 @@
+import Vuex from 'vuex'
+import DesignSystem from 'owncloud-design-system'
+import stubs from '@/tests/unit/stubs/index.js'
+import { createLocalVue, mount } from '@vue/test-utils'
+import FileActions from '@files/src/components/SideBar/Actions/FileActions.vue'
+
+import GetTextPlugin from 'vue-gettext'
+
+import {
+  apps,
+  getActions,
+  basicActions,
+  extraActions
+} from '@/tests/unit/__fixtures__/fileActions.js'
+
+const localVue = createLocalVue()
+localVue.use(DesignSystem)
+localVue.use(Vuex)
+localVue.use(GetTextPlugin, {
+  translations: 'does-not-matter.json',
+  silent: true
+})
+
+const filesPersonalRoute = { name: 'files-personal' }
+
+function getWrapper(route, { filename = 'testfile', extension, extraActions = {}, type = 'file' }) {
+  const mountStubs = { ...stubs, 'oc-button': false }
+  const mountOptions = {
+    localVue,
+    store: createStore({}, filename, extension, type),
+    stubs: mountStubs,
+    mocks: {
+      $route: route,
+      publicPage: () => false
+    },
+    computed: {
+      actions() {
+        let actions = getActions(extraActions)
+        if (type === 'folder') {
+          actions = actions.filter(item => item.icon !== 'download')
+        }
+        return actions
+      }
+    }
+  }
+  return mount(FileActions, mountOptions)
+}
+
+describe('FileActions', () => {
+  describe('when user is on personal route', () => {
+    afterEach(() => {
+      jest.clearAllMocks()
+    })
+
+    it('should show action buttons on md file', async () => {
+      const actions = { ...basicActions, 'markdown-editor': extraActions['markdown-editor'] }
+      const wrapper = getWrapper(filesPersonalRoute, {
+        filename: 'welcome',
+        extension: 'md',
+        extraActions: ['markdown-editor']
+      })
+
+      for (const button in actions) {
+        const action = actions[button]
+
+        const buttonElement = wrapper.find(action.selector)
+        expect(buttonElement.exists()).toBeTruthy()
+
+        await buttonElement.trigger('click.stop')
```

Comment:
From this line until the end of the test scenario goes into the direction of an integration test. The fact that the button exists is a good unit test and well done. The interaction with the button is way out of scope for this unit test. Same for the other tests in this spec file.

Reply:
Well, that's unfortunate, We have tested click on this and many other components. We did so because the original issue has mentioned testing events such as click, input etc. 
https://github.com/owncloud/web/issues/5217
Shall we refactor the integration tests into a separate file of remove them altogether?

Reply:
Don't get us wrong, the work done is great 😉 and unit tests _should_ test clicks and other events, it's just that the results of those clicks are tested a bit too deeply in here (and go out of scope of the component, basically)

Could you try refactoring this into an integration test (in a separate PR and remove it from this one)?

Reply:
What about only checking here just that the action was executed `expect(action.handler).toHaveBeenCalledTimes(1)` and move the rests to the integration tests?